unit_tests/test_size_original_value.py

```diff
@@ -116,7 +116,7 @@ def result_equals_render_after_reset(context):
     assert context.result == context.uut.render()
 
 
-@then('Result equals .render() after Request reset()')
+@then('Result equals .render() after Request reset()')  # noqa: F811
```

Comment:
It's peculiar that these are required only on these two declarations. 🤷‍♂

Reply:
My guess is that the check just runs from top to bottom and looks for redefinitions.
Would you rather have the ignore on both functions?
Or shall we fix this by simply renaming one of the functions?

Reply:
Oh, the function name is reused... Yes, simply renaming the functions seems like the intuitive fix.

This is easy to do while using the BDD style, since the function tag matters more than the function name. Having warnings enabled will hopefully make it easier to see when new warnings are added in the future.

Reply:
I've renamed the functions but it does seem to have an effect on which of the two functions with the same name is called.
This doesn't change the outcome in the case of `result_equals_render_after_reset`, but does for `request_one_block`.
It appears that the redefinition overwrote `@given('A Size')` completely so that the test was always using the second function with `@given('A Size whose target block will change size upon mutation')`
Both function tags seem to point at the second, redefined function name.
As with the renaming it uses the intended function now, the unit tests fail because of an endless recursion.

I don't know how to quickly fix this so we either leave it as it was running the whole time by using the `@given('A Size whose target block will change size upon mutation')` for all scenarios or we fix the error it produces.
@jtpereyda feel free to add commits here to fix this!

Reply:
@jtpereyda 0aafd5e might fix the problem.
I took a look at `request_one_block_with_size_change` and noticed, that `request.pop()` is called before defining the size block. So I applied the same structure on `request_one_block` and now the test passes.

I didn't really verify that the check does exactly what it's supposed to do, would be nice if you could double-check.

Reply:
@SR4ven I think your change is in line with the purpose of the test, as the text doesn't seem to describe any recursive behavior. I don't know if there's any unit test for Size under recursive conditions though, which is a bit of a gap. Seems like this change is good to go for now anyway.